tools/kubectl-karpenter/cmd/kubectl-karpenter/main.go

```diff
@@ -0,0 +1,44 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+
+	kk "github.com/aws/karpenter/tools/kubectl-karpenter/pkg/kubectl-karpenter"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/component-base/cli"
+)
+
+func main() {
+	flags := pflag.NewFlagSet("kubectl-karpenter", pflag.ExitOnError)
+	pflag.CommandLine = flags
+
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
```

Comment:
I didn't read through the `kubectl` upstream code details too much. Can you just give me the high-level overview of the value of all of these commends? Is there anything that we don't need here and we could pare down?

Reply:
I did a cleanup on this - many of those options were taken from kubectl-convert which can contact the API server and get the resources from there. Our tool doesn't need those capabilities, so I cleaned it up